templates/app/flake.nix

```diff
@@ -0,0 +1,34 @@
+{
+  description = "Application packaged using poetry2nix";
+
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs";
+  inputs.poetry2nix.url = "github:nix-community/poetry2nix";
+
+  outputs = { self, nixpkgs, flake-utils, poetry2nix }:
+    {
+      # Nixpkgs overlay providing the application
+      overlay = nixpkgs.lib.composeManyExtensions [
```

Comment:
This part I don't like very much, as I think this can lead to the multiple inheritance diamond problem. The alternative is to have the overlay rely on the `poetry2nix` that is part of Nixpkgs, in which case `poetry2nix.overlay` is removed here. That I don't like at all, because the user of this template explicitly points to the `poetry2nix` repo.

However, further down, where the applications are defined, I think it *is

Reply:
I think I am missing the idea here but can't we set `overlays` instead ? e.g. why do we need composeManyExtensions

Reply:
Overlays are not propagated. Consider A depends on B, and B on C. B can use C as overlay for itself. If you now want to use B in A through overlays, then you need to use the overlays that B in order to use B.